src/net_processing.cpp

```diff
@@ -1422,45 +1422,35 @@ void PeerLogicValidation::BlockChecked(const CBlock& block, const BlockValidatio
 
 bool static AlreadyHave(const CInv& inv, const CTxMemPool& mempool) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 {
-    switch (inv.type)
-    {
-    case MSG_TX:
-    case MSG_WITNESS_TX:
-    case MSG_WTX:
+    if (inv.IsGenTxMsg()) {
+        assert(recentRejects);
+        if (::ChainActive().Tip()->GetBlockHash() != hashRecentRejectsChainTip) {
+            // If the chain tip has changed previously rejected transactions
+            // might be now valid, e.g. due to a nLockTime'd tx becoming valid,
+            // or a double-spend. Reset the rejects filter and give those
+            // txs a second chance.
+            hashRecentRejectsChainTip = ::ChainActive().Tip()->GetBlockHash();
+            recentRejects->reset();
+        }
         {
-            assert(recentRejects);
-            if (::ChainActive().Tip()->GetBlockHash() != hashRecentRejectsChainTip)
-            {
-                // If the chain tip has changed previously rejected transactions
-                // might be now valid, e.g. due to a nLockTime'd tx becoming valid,
-                // or a double-spend. Reset the rejects filter and give those
-                // txs a second chance.
-                hashRecentRejectsChainTip = ::ChainActive().Tip()->GetBlockHash();
-                recentRejects->reset();
-            }
-
-            {
-                LOCK(g_cs_orphans);
-                if (!inv.IsMsgWtx() && mapOrphanTransactions.count(inv.hash)) {
-                    return true;
-                } else if (inv.IsMsgWtx() && g_orphans_by_wtxid.count(inv.hash)) {
-                    return true;
-                }
-            }
-
-            {
-                LOCK(g_cs_recent_confirmed_transactions);
-                if (g_recent_confirmed_transactions->contains(inv.hash)) return true;
+            LOCK(g_cs_orphans);
+            if (!inv.IsMsgWtx() && mapOrphanTransactions.count(inv.hash)) {
+                return true;
+            } else if (inv.IsMsgWtx() && g_orphans_by_wtxid.count(inv.hash)) {
+                return true;
```

Comment:
nit: this was the code before, but since you are re-indenting/modifying it:

```cpp
if (A) {
    C;
} else if (B) {
    C;
}
```

is the same as
```
if (A || B) {
   C;
}
```

Reply:
Yes, I deliberately did not touch it to keep the concepts separate.